### PR TITLE
Add AVBR, QVBR, ICQ to sample_multi_transcode

### LINKED ISSUE #1641: Add ICQ, AVBR and QVBR to smt?
HI, its possible to integrate in sample_multi_transcode ICQ, AVBR and QVBR RC modes?

### PR DESCRIPTION
**Pull request by aobolensk**
Add flags for selection of AVBR, QVBR, ICQ rate control methods to sample_multi_transcode.
Closes #1641

### CODE REVIEW

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 1]
```diff
@@ -238,6 +238,9 @@ void TranscodingSample::PrintHelp()
     msdk_printf(MSDK_STRING("                May be 1 in the case when -mss option is specified \n"));
     msdk_printf(MSDK_STRING("  -la_ext       Use external LA plugin (compatible with h264 & hevc encoders)\n"));
     msdk_printf(MSDK_STRING("  -vbr          Variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -avbr         Average variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -qvbr         Quality-defined variable bitrate control\n"));
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: Doesn't it also need ICQQuality and QVBRQuality parameters?

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 2]
```diff
@@ -1205,6 +1210,30 @@ mfxStatus ParseAdditionalParams(msdk_char *argv[], mfxU32 argc, mfxU32& i, Trans
         InputParams.bPrefferdGfx = true;
     }
 #endif
+    else if (0 == msdk_strcmp(argv[i], MSDK_STRING("-avbr")))
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: AVBR requires additional parameters.

**Comment by dmitryermilov** [comment]
@aobolensk , please reply

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 3]
```diff
@@ -238,6 +238,13 @@ void TranscodingSample::PrintHelp()
     msdk_printf(MSDK_STRING("                May be 1 in the case when -mss option is specified \n"));
     msdk_printf(MSDK_STRING("  -la_ext       Use external LA plugin (compatible with h264 & hevc encoders)\n"));
     msdk_printf(MSDK_STRING("  -vbr          Variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -avbr         Average variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -convergence  bitrate convergence period for avbr, in the unit of frame\n"));
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: If name is followed by value, the value holder like <value> should be in the help string.

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 4]
```diff
@@ -238,6 +238,13 @@ void TranscodingSample::PrintHelp()
...
+    msdk_printf(MSDK_STRING("  -accuracy     bitrate accuracy for avbr, in the range of [1, 100]\n"));
+    msdk_printf(MSDK_STRING("  -qvbr <quality>\n"));
+    msdk_printf(MSDK_STRING("                Quality-defined variable bitrate control\n"));
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: Please provide the range here and below if possible.

File: samples/sample_multi_transcode/src/pipeline_transcode.cpp [hunk 5]
```diff
@@ -2337,6 +2337,19 @@ mfxStatus CTranscodingPipeline::InitEncMfxParams(sInputParams *pInParams)
...
+        m_mfxEncParams.mfx.Accuracy = pInParams->Accuracy;
+        m_mfxEncParams.mfx.Convergence = pInParams->Convergence;
+    }
+    else if (pInParams->nRateControlMethod == MFX_RATECONTROL_QVBR)
+    {
+        m_mfxEncParams.mfx.Quality = pInParams->QVBRQuality;
+    }
+    else if (pInParams->nRateControlMethod == MFX_RATECONTROL_ICQ)
+    {
+        m_mfxEncParams.mfx.Quality = pInParams->ICQQuality;
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/pipeline_transcode.cpp]: It should be mfx.ICQQuality

File: samples/sample_multi_transcode/src/pipeline_transcode.cpp [hunk 6]
```diff
@@ -2337,6 +2337,19 @@ mfxStatus CTranscodingPipeline::InitEncMfxParams(sInputParams *pInParams)
...
         m_mfxEncParams.mfx.QPB = pInParams->nQPB;
     }
+    else if (pInParams->nRateControlMethod == MFX_RATECONTROL_AVBR)
+    {
+        m_mfxEncParams.mfx.Accuracy = pInParams->Accuracy;
+        m_mfxEncParams.mfx.Convergence = pInParams->Convergence;
+    }
+    else if (pInParams->nRateControlMethod == MFX_RATECONTROL_QVBR)
+    {
+        m_mfxEncParams.mfx.Quality = pInParams->QVBRQuality;
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/pipeline_transcode.cpp]: It should be CO3.QVBRQuality

File: samples/sample_multi_transcode/src/transcode_utils.cpp (hunk 3 above)
Comment (aobolensk) [reply samples/sample_multi_transcode/src/transcode_utils.cpp]: Done

File: samples/sample_multi_transcode/src/transcode_utils.cpp (hunk 4 above)
Comment (aobolensk) [reply samples/sample_multi_transcode/src/transcode_utils.cpp]: Done

File: samples/sample_multi_transcode/src/pipeline_transcode.cpp (hunk 5 above)
Comment (aobolensk) [reply samples/sample_multi_transcode/src/pipeline_transcode.cpp]: Done

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 7]
```diff
@@ -238,6 +238,15 @@ void TranscodingSample::PrintHelp()
...
     msdk_printf(MSDK_STRING("  -vbr          Variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -avbr         Average variable bitrate control\n"));
+    msdk_printf(MSDK_STRING("  -convergence <number>\n"));
+    msdk_printf(MSDK_STRING("                bitrate convergence period for avbr, in the unit of frame, range: [1, 65535]\n"));
+    msdk_printf(MSDK_STRING("  -accuracy <number>\n"));
+    msdk_printf(MSDK_STRING("                bitrate accuracy for avbr, in range: [1, 65535]\n"));
+    msdk_printf(MSDK_STRING("  -qvbr <quality>\n"));
+    msdk_printf(MSDK_STRING("                Quality-defined variable bitrate control, quality is in range of: [0, 100]\n"));
+    msdk_printf(MSDK_STRING("  -icq <quality>\n"));
+    msdk_printf(MSDK_STRING("                Intelligent constant quality bitrate control, quality is in range of: [0, 100]\n"));
```
Comment (lakulako) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: Please take descriptions and values from mediasdk_man

Comment (aobolensk) [reply samples/sample_multi_transcode/src/transcode_utils.cpp]: Fixed descriptions

File: samples/sample_multi_transcode/src/pipeline_transcode.cpp (hunk 6 above)
Comment (aobolensk) [reply samples/sample_multi_transcode/src/pipeline_transcode.cpp]: Done

**Pull request merged**